docs/topics/con_configmap-availability.adoc

```diff
@@ -0,0 +1,145 @@
+pool.size.max: 16
+----
+
+* If your ConfigMap contains a property named `application.yaml`, {deploymentPlatform} applies the content of that property as an embedded YAML application template.
+This is useful in case entire sets of properties need to be updated at once.
+It also helps make complex templates more readable.
++
+[source,yaml,options="nowrap"]
+----
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: demo
+data:
+  application.yaml: |-
+    pool:
+      size:
+        core: 1
+        max:16
+----
+
+* If your ConfigMap contains a property named `application.properties`, the content of that property is treated as an embedded `application.properties` file:
+This approach is useful for applications that use multiple profiles, where the configurations differ depending on the active profile.
+This structure allows you to maintain multiple parallel configurations in a manner similar to maintaining multiple property files.
+In the example below, alternate configurations are separated by the `---` sequence of characters.
++
+[source,yaml,options="nowrap"]
+----
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: demo
+data:
+  application.yml: |-
+    greeting:
+      message: Say Hello to the World
+    ---
+    spring:
+      profiles: development
+    greeting:
+      message: Say Hello to the Developers
+    ---
+    spring:
+      profiles: production
+    greeting:
+      message: Say Hello to the Ops
+----
+
+////
+Spring Boot applications can also be configured differently depending on active profiles and it is possible to
+provide different property values for different profiles using an `application.properties|yaml` property, specifying
+profile-specific values each in their own document (indicated by the `---` sequence) as follows:
+////
+
+* Use a system property to select which profile to enable when you run your application.
++
+. Store the system property in the `JAVA_OPTIONS` environment variable in the deployment configuration of your application.
+//`DeploymentConfig` or Kubernetes `ReplicationConfig` resource file as follows:
++
+[source,yaml,options="nowrap"]
+----
+apiVersion: v1
+kind: DeploymentConfig
+spec:
+  replicas: 1
+  ...
+    spec:
+      containers:
+      - env:
+        - name: JAVA_APP_DIR
+          value: /deployments
+        - name: JAVA_OPTIONS
+          value: -Dspring.profiles.active=developer
+----
++
+. Pass in the system property when starting your application to determine what profile is used.
+To use the `developer` profile, for example:
++
+[source,bash,options="nowrap"]
+----
+java -jar {app-name}.jar -Dspring.profiles.active=developer
+----
+
+// make this a prerequisite:
+**Notes:**
+* You must assign view permissions to your OpenShift service account to access ConfigMap on OpenShift the service account needs at least view permissions i.e.:
++
+[source,bash,options="nowrap"]
+----
+oc policy add-role-to-user view system:serviceaccount:$(oc project -q):default -n $(oc project -q)
```

Comment:
Recommended practice, I think, is to have a dedicated service account for each application, not use the `default` one.